packages/graphql-key-transformer/src/KeyTransformer.ts

```diff
@@ -963,3 +977,245 @@ function addCompositeSortKey(
+          ref(`${queryExprReference}.expression`),
+          raw(`"$${queryExprReference}.expression AND #sortKey BETWEEN :sortKey0 AND :sortKey1"`),
+        ),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey0", { "$SkType" : $ctx.args.get($sk).between[0] })`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey1", { "$SkType": $ctx.args.get($sk).between[1] })`),
+      ]),
+    ),
+    iff(
+      raw(`!$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).eq)`),
+      compoundExpression([
+        set(ref(`${queryExprReference}.expression`), raw(`"$${queryExprReference}.expression AND #sortKey = :sortKey"`)),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey", { "$SkType": $ctx.args.get($sk).eq })`),
+      ]),
+    ),
+    iff(
+      raw(`!$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).lt)`),
+      compoundExpression([
+        set(ref(`${queryExprReference}.expression`), raw(`"$${queryExprReference}.expression AND #sortKey < :sortKey"`)),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey", { "$SkType": $ctx.args.get($sk).lt })`),
+      ]),
+    ),
+    iff(
+      raw(`!$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).le)`),
+      compoundExpression([
+        set(ref(`${queryExprReference}.expression`), raw(`"$${queryExprReference}.expression AND #sortKey <= :sortKey"`)),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).le })`),
+      ]),
+    ),
+    iff(
+      raw(`!$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).gt)`),
+      compoundExpression([
+        set(ref(`${queryExprReference}.expression`), raw(`"$${queryExprReference}.expression AND #sortKey > :sortKey"`)),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).gt })`),
+      ]),
+    ),
+    iff(
+      raw(`!$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).ge)`),
+      compoundExpression([
+        set(ref(`${queryExprReference}.expression`), raw(`"$${queryExprReference}.expression AND #sortKey >= :sortKey"`)),
+        qref(`$${queryExprReference}.expressionNames.put("#sortKey", $sk)`),
+        qref(`$${queryExprReference}.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).ge })`),
+      ]),
+    ),
+  ]);
+}
+
+function makeSyncQueryResolver() {
+  const requestVariable = 'QueryRequest';
+  const expressions: Expression[] = [];
+  expressions.push(
+    ifElse(
+      raw('!$scan'),
```

Comment:
In the else statement it's still creating a sync operation payload.

Reply:
yeah , if no (pk,sk) matches in VTL , the operation will be a scan operation rather than query.

Reply:
Can we bubble up the logic on where to add query related key/value pairs above the payload? (such as queryExp, index, and scanIndexForward)